os_brick/initiator/connectors/lightos.py

```diff
@@ -62,6 +62,23 @@ def __init__(self,
         self.message_queue = message_queue
         self.DISCOVERY_DIR_PATH = '/etc/discovery-client/discovery.d/'
 
+    @staticmethod
+    def get_ip_addresses():
+        ip_addresses = []
+        result = subprocess.check_output(['ip', 'addr']).decode('utf-8')
+        lines = result.split('\n')
+        for line in lines:
+          if 'inet ' in line:
+              parts = line.split()
+              ip_parts = parts[1]
+              ip_parts = ip_parts.split('/')
+              host_ip=ip_parts[0]
+              if '127.0.0.1' in host_ip:
+                 continue
+              ip_addresses.append(host_ip)
```

Comment:
I am not sure about the security implications here -- can you make a convincing argument that all IP addresses the host has (even if internal or tunnel or VPN or ...) should be added to the volume IPACL? I think if you look at the ticket you will see a list of addresses @sagi-lb recommended to filter out.

Reply:
This returns all IPv4 addresses and writes to volume IP-ACL,  also return IPs when VPN is connected,  I tested on my local machine when i connect via VPN/tunnel to lab environment. I've not tested IPv6 and need to add support for  IPv6.

Reply:
See @sagi-lb comment here https://lightbitslabs.atlassian.net/browse/LBM1-28409?focusedCommentId=113743 for which addresses to filter

Reply:
I'll apply filters

Reply:
I hope that using `netifaces` (see previous comment) it will be easier

Reply:
netifaces works perfectly to get IPv4 and IPv6 address, trying to find if an address is physical or virtual for the filtering purposes.

Reply:
I highly suggest using existing openstack tools like:
https://github.com/openstack/oslo.utils/blob/cecf061e6e4f666104170e3bba93fdae24255309/oslo_utils/netutils.py#L369

Reply:
FWIW I 100% agree with @yuval-lb . If there is an openstack utility that does what we need, it's best to use it. It will make upstream submission much easier.

Reply:
@muliby-lb @yuval-lb
Interestingly netutil.py is also using package netifaces, which means we should not have issues using netifaces.
I'm using netutil.py for IPv4/6 validity checks. Current code will read all available IPs and pass-on to cinder.
If IPv6 is not enabled on host then this function will read only read IPv4 addresses.
It will not be a bad idea to move forward with all IPs and then apply filtering on undesired IPs once everything works.

@sagi-lb Is there a way to determine if an IP belongs to a physical interface vs virtual.

Reply:
oslo_utils/netutils.py has function that returns only one IP,  but not all IPs. In a multi IPs environment this will become a problem.